actionpack/lib/action_dispatch/middleware/session/abstract_store.rb

```diff
@@ -3,18 +3,22 @@
 require 'rack/session/abstract/id'
 require 'action_dispatch/middleware/cookies'
 require 'action_dispatch/request/session'
+require "active_support/core_ext/module/delegation"
 
 module ActionDispatch
   module Session
     class SessionRestoreError < StandardError #:nodoc:
-      attr_reader :original_exception
-
-      def initialize(const_error)
-        @original_exception = const_error
+      delegate :backtrace, to: :cause
 
+      def initialize(message)
         super("Session contains objects whose class definition isn't available.\n" +
           "Remember to require the classes for all objects kept in the session.\n" +
-          "(Original exception: #{const_error.message} [#{const_error.class}])\n")
+          "(Original exception: #{message})\n")
```

Comment:
Unfortunately, #cause returns nil in #initialize. What do you think of using $!?